### ISSUE TITLE
💅 False positive: import reported as "only used as types" when used in Svelte template

### ISSUE THREAD

**Issue by bennajah** [biomejs/biome#9358]
### Environment information

<details>

```bash

CLI:
  Version:                      2.4.6
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_PATH:                    unset
  BIOME_LOG_PREFIX_NAME:             unset
  BIOME_LOG_LEVEL:                   unset
  BIOME_LOG_KIND:                    unset
  BIOME_CONFIG_PATH:                 unset
  BIOME_THREADS:                     unset
  BIOME_WATCHER_KIND:                unset
  BIOME_WATCHER_POLLING_INTERVAL:    unset
  NO_COLOR:                     unset
  TERM:                         unset
  JS_RUNTIME_VERSION:           v24.14.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true
  HTML full support enabled:    true

Linter:
  JavaScript enabled:           unset
  JSON enabled:                 unset
  CSS enabled:                  unset
  GraphQL enabled:              unset
  Recommended:                  unset
  Enabled rules:
    a11y/noAccessKey
    a11y/noAriaHiddenOnFocusable
    a11y/noAriaUnsupportedElements
    a11y/noDistractingElements
    a11y/noHeaderScope
    a11y/noInteractiveElementToNoninteractiveRole
    a11y/noLabelWithoutControl
    a11y/noNoninteractiveElementInteractions
    a11y/noNoninteractiveElementToInteractiveRole
    a11y/noNoninteractiveTabindex
    a11y/noPositiveTabindex
    a11y/noRedundantAlt
    a11y/noRedundantRoles
    a11y/noStaticElementInteractions
    a11y/noSvgWithoutTitle
    a11y/useAltText
    a11y/useAnchorContent
    a11y/useAriaActivedescendantWithTabindex
    a11y/useAriaPropsForRole
    a11y/useAriaPropsSupportedByRole
    a11y/useButtonType
    a11y/useFocusableInteractive
    a11y/useGenericFontNames
    a11y/useHeadingContent
    a11y/useHtmlLang
    a11y/useIframeTitle
    a11y/useKeyWithClickEvents
    a11y/useKeyWithMouseEvents
    a11y/useMediaCaption
    a11y/useSemanticElements
    a11y/useValidAnchor
    a11y/useValidAriaProps
    a11y/useValidAriaRole
    a11y/useValidAriaValues
    a11y/useValidAutocomplete
    a11y/useValidLang
    complexity/noAdjacentSpacesInRegex
    complexity/noArguments
    complexity/noBannedTypes
    complexity/noCommaOperator
    complexity/noEmptyTypeParameters
    complexity/noExcessiveCognitiveComplexity
    complexity/noExcessiveNestedTestSuites
    complexity/noExtraBooleanCast
    complexity/noFlatMapIdentity
    complexity/noForEach
    complexity/noStaticOnlyClass
    complexity/noThisInStatic
    complexity/noUselessCatch
    complexity/noUselessConstructor
    complexity/noUselessContinue
    complexity/noUselessEmptyExport
    complexity/noUselessEscapeInRegex
    complexity/noUselessFragments
    complexity/noUselessLabel
    complexity/noUselessLoneBlockStatements
    complexity/noUselessRename
    complexity/noUselessStringConcat
    complexity/noUselessStringRaw
    complexity/noUselessSwitchCase
    complexity/noUselessTernary
    complexity/noUselessThisAlias
    complexity/noUselessTypeConstraint
    complexity/noUselessUndefinedInitialization
    complexity/noVoid
    complexity/useArrowFunction
    complexity/useDateNow
    complexity/useFlatMap
    complexity/useIndexOf
    complexity/useLiteralKeys
    complexity/useNumericLiterals
    complexity/useOptionalChain
    complexity/useRegexLiterals
    complexity/useSimpleNumberKeys
    complexity/useSimplifiedLogicExpression
    complexity/useWhile
    correctness/noConstAssign
    correctness/noConstantCondition
    correctness/noConstantMathMinMaxClamp
    correctness/noConstructorReturn
    correctness/noEmptyCharacterClassInRegex
    correctness/noEmptyPattern
    correctness/noGlobalDirnameFilename
    correctness/noGlobalObjectCalls
    correctness/noInnerDeclarations
    correctness/noInvalidBuiltinInstantiation
    correctness/noInvalidConstructorSuper
    correctness/noInvalidDirectionInLinearGradient
    correctness/noInvalidGridAreas
    correctness/noInvalidPositionAtImportRule
    correctness/noInvalidUseBeforeDeclaration
    correctness/noMissingVarFunction
    correctness/noNonoctalDecimalEscape
    correctness/noPrecisionLoss
    correctness/noRestrictedElements
    correctness/noSelfAssign
    correctness/noSetterReturn
    correctness/noStringCaseMismatch
    correctness/noSwitchDeclarations
    correctness/noUndeclaredVariables
    correctness/noUnknownFunction
    correctness/noUnknownMediaFeatureName
    correctness/noUnknownProperty
    correctness/noUnknownPseudoClass
    correctness/noUnknownPseudoElement
    correctness/noUnknownTypeSelector
    correctness/noUnknownUnit
    correctness/noUnmatchableAnbSelector
    correctness/noUnreachable
    correctness/noUnreachableSuper
    correctness/noUnsafeFinally
    correctness/noUnsafeOptionalChaining
    correctness/noUnusedFunctionParameters
    correctness/noUnusedImports
    correctness/noUnusedLabels
    correctness/noUnusedPrivateClassMembers
    correctness/noUnusedVariables
    correctness/noVoidElementsWithChildren
    correctness/noVoidTypeReturn
    correctness/useGraphqlNamedOperations
    correctness/useImageSize
    correctness/useIsNan
    correctness/useParseIntRadix
    correctness/useSingleJsDocAsterisk
    correctness/useValidForDirection
    correctness/useValidTypeof
    correctness/useYield
    nursery/useSortedClasses
    performance/noAccumulatingSpread
    performance/noBarrelFile
    performance/noDelete
    performance/noDynamicNamespaceImportAccess
    performance/noNamespaceImport
    performance/noUnwantedPolyfillio
    performance/useGoogleFontPreconnect
    performance/useTopLevelRegex
    security/noBlankTarget
    security/noDangerouslySetInnerHtml
    security/noDangerouslySetInnerHtmlWithChildren
    security/noGlobalEval
    style/noDescendingSpecificity
    style/noDoneCallback
    style/noEnum
    style/noExportedImports
    style/noInferrableTypes
    style/noNamespace
    style/noNegationElse
    style/noNestedTernary
    style/noNonNullAssertion
    style/noParameterAssign
    style/noParameterProperties
    style/noRestrictedGlobals
    style/noRestrictedImports
    style/noRestrictedTypes
    style/noShoutyConstants
    style/noSubstr
    style/noUnusedTemplateLiteral
    style/noUselessElse
    style/noValueAtRule
    style/noYodaExpression
    style/useArrayLiterals
    style/useAsConstAssertion
    style/useAtIndex
    style/useBlockStatements
    style/useCollapsedElseIf
    style/useCollapsedIf
    style/useConsistentArrayType
    style/useConsistentBuiltinInstantiation
    style/useConsistentMemberAccessibility
    style/useConsistentObjectDefinitions
    style/useConsistentTypeDefinitions
    style/useConst
    style/useDefaultParameterLast
    style/useDefaultSwitchClause
    style/useDeprecatedReason
    style/useEnumInitializers
    style/useExponentiationOperator
    style/useExportType
    style/useFilenamingConvention
    style/useForOf
    style/useFragmentSyntax
    style/useGraphqlNamingConvention
    style/useGroupedAccessorPairs
    style/useImportType
    style/useLiteralEnumMembers
    style/useNodeAssertStrict
    style/useNodejsImportProtocol
    style/useNumberNamespace
    style/useNumericSeparators
    style/useObjectSpread
    style/useReadonlyClassProperties
    style/useSelfClosingElements
    style/useShorthandAssign
    style/useShorthandFunctionType
    style/useSymbolDescription
    style/useTemplate
    style/useThrowNewError
    style/useThrowOnlyError
    style/useTrimStartEnd
    style/useUnifiedTypeSignatures
    suspicious/noAlert
    suspicious/noApproximativeNumericConstant
    suspicious/noAssignInExpressions
    suspicious/noAsyncPromiseExecutor
    suspicious/noBiomeFirstException
    suspicious/noBitwiseOperators
    suspicious/noCatchAssign
    suspicious/noClassAssign
    suspicious/noCommentText
    suspicious/noCompareNegZero
    suspicious/noConfusingLabels
    suspicious/noConfusingVoidType
    suspicious/noConstEnum
    suspicious/noConstantBinaryExpressions
    suspicious/noControlCharactersInRegex
    suspicious/noDebugger
    suspicious/noDocumentCookie
    suspicious/noDoubleEquals
    suspicious/noDuplicateAtImportRules
    suspicious/noDuplicateCase
    suspicious/noDuplicateClassMembers
    suspicious/noDuplicateCustomProperties
    suspicious/noDuplicateElseIf
    suspicious/noDuplicateFields
    suspicious/noDuplicateFontNames
    suspicious/noDuplicateJsxProps
    suspicious/noDuplicateObjectKeys
    suspicious/noDuplicateParameters
    suspicious/noDuplicateProperties
    suspicious/noDuplicateSelectorsKeyframeBlock
    suspicious/noDuplicateTestHooks
    suspicious/noEmptyBlock
    suspicious/noEmptyBlockStatements
    suspicious/noEmptyInterface
    suspicious/noEvolvingTypes
    suspicious/noExplicitAny
    suspicious/noExportsInTest
    suspicious/noExtraNonNullAssertion
    suspicious/noFallthroughSwitchClause
    suspicious/noFocusedTests
    suspicious/noFunctionAssign
    suspicious/noGlobalAssign
    suspicious/noGlobalIsFinite
    suspicious/noGlobalIsNan
    suspicious/noImplicitAnyLet
    suspicious/noImportAssign
    suspicious/noImportantInKeyframe
    suspicious/noIrregularWhitespace
    suspicious/noLabelVar
    suspicious/noMisleadingCharacterClass
    suspicious/noMisleadingInstantiator
    suspicious/noMisplacedAssertion
    suspicious/noMisrefactoredShorthandAssign
    suspicious/noNonNullAssertedOptionalChain
    suspicious/noOctalEscape
    suspicious/noPrototypeBuiltins
    suspicious/noQuickfixBiome
    suspicious/noRedeclare
    suspicious/noRedundantUseStrict
    suspicious/noSelfCompare
    suspicious/noShadowRestrictedNames
    suspicious/noShorthandPropertyOverrides
    suspicious/noSkippedTests
    suspicious/noSparseArray
    suspicious/noSuspiciousSemicolonInJsx
    suspicious/noTemplateCurlyInString
    suspicious/noThenProperty
    suspicious/noTsIgnore
    suspicious/noUnassignedVariables
    suspicious/noUnknownAtRules
    suspicious/noUnsafeDeclarationMerging
    suspicious/noUnsafeNegation
    suspicious/noUselessEscapeInString
    suspicious/noUselessRegexBackrefs
    suspicious/noVar
    suspicious/noWith
    suspicious/useAdjacentOverloadSignatures
    suspicious/useAwait
    suspicious/useBiomeIgnoreFolder
    suspicious/useDefaultSwitchClauseLast
    suspicious/useErrorMessage
    suspicious/useGetterReturn
    suspicious/useGoogleFontDisplay
    suspicious/useGuardForIn
    suspicious/useIsArray
    suspicious/useIterableCallbackReturn
    suspicious/useNamespaceKeyword
    suspicious/useNumberToFixedDigitsArgument
    suspicious/useStaticResponseMethods
    suspicious/useStrictMode

Workspace:
  Open Documents:               0

```

</details>

### Rule name

lint/style/useImportType

### Playground link

https://biomejs.dev/playground/?indentScriptAndStyle=true&files.main.svelte=PABzAGMAcgBpAHAAdAAgAGwAYQBuAGcAPQAiAHQAcwAiAD4ACgAJAGkAbQBwAG8AcgB0ACAAewAgAGIAdQB0AHQAbwBuAEMAbABhAHMAcwAgAH0AIABmAHIAbwBtACAAIgAuAC8AYgB1AHQAdABvAG4AQwBsAGEAcwBzAC4AdABzACIAOwAKAAkAZQB4AHAAbwByAHQAIABsAGUAdAAgAHYAYQByAGkAYQBuAHQAOgAgAFAAYQByAGEAbQBlAHQAZQByAHMAPAB0AHkAcABlAG8AZgAgAGIAdQB0AHQAbwBuAEMAbABhAHMAcwA%2BAFsAMABdACAAPQAgACIAZABlAGYAYQB1AGwAdAAiADsACgA8AC8AcwBjAHIAaQBwAHQAPgAKAAoAPABiAHUAdAB0AG8AbgAgAHQAeQBwAGUAPQAiAGIAdQB0AHQAbwBuACIAIABjAGwAYQBzAHMAPQB7AGIAdQB0AHQAbwBuAEMAbABhAHMAcwAoAHYAYQByAGkAYQBuAHQAKQB9AD4AQgB1AHQAdABvAG4APAAvAGIAdQB0AHQAbwBuAD4ACgA%3D&files.buttonClass.ts=ZQB4AHAAbwByAHQAIABmAHUAbgBjAHQAaQBvAG4AIABiAHUAdAB0AG8AbgBDAGwAYQBzAHMAKAB2AGEAcgBpAGEAbgB0ADoAIAAiAGQAZQBmAGEAdQBsAHQAIgAgAHwAIAAiAHMAZQBjAG8AbgBkAGEAcgB5ACIAIAA9ACAAIgBkAGUAZgBhAHUAbAB0ACIAKQAgAHsACgAJAGMAbwBuAHMAdAAgAGIAYQBzAGUAIAA9ACAAIgBwAHgALQA0ACAAcAB5AC0AMgAgAHIAbwB1AG4AZABlAGQAIABmAG8AbgB0AC0AYgBvAGwAZAAiADsACgAJAGMAbwBuAHMAdAAgAHYAYQByAGkAYQBuAHQAQwBsAGEAcwBzACAAPQAKAAkACQB2AGEAcgBpAGEAbgB0ACAAPQA9AD0AIAAiAHMAZQBjAG8AbgBkAGEAcgB5ACIACgAJAAkACQA%2FACAAIgBiAGcALQBnAHIAYQB5AC0AMgAwADAAIAB0AGUAeAB0AC0AYgBsAGEAYwBrACIACgAJAAkACQA6ACAAIgBiAGcALQBiAGwAdQBlAC0ANQAwADAAIAB0AGUAeAB0AC0AdwBoAGkAdABlACIAOwAKAAkAcgBlAHQAdQByAG4AIABgACQAewBiAGEAcwBlAH0AIAAkAHsAdgBhAHIAaQBhAG4AdABDAGwAYQBzAHMAfQBgADsACgB9AAoA

### Expected result

**Steps to Reproduce**

1. Create a runtime function buttonClass.ts:
```ts
// buttonClass.ts
export function buttonClass(variant: "default" | "secondary" = "default") {
  return variant === "secondary" ? "bg-gray-200" : "bg-blue-500";
}
```
2. Create a Svelte component Button.svelte:
```svelte
<script lang="ts">
  import { buttonClass } from "./buttonClass.ts";

  export let variant: Parameters<typeof buttonClass>[0] = "default";
</script>

<button class={buttonClass(variant)}>Test</button>
```

**Expected Result**

Biome should not report any warning because buttonClass is actually used at runtime in the template.

The component is valid TypeScript and Svelte 5.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
